probe-rs/src/architecture/arm/sequences.rs

```diff
@@ -551,14 +555,15 @@ pub trait ArmDebugSequence: Send + Sync + Debug {
             // End of atomic block.
 
             // SWD or JTAG should now be activated, so we can try and connect to the debug port.
-            if self.debug_port_connect(interface, dp).is_ok() {
-                return Ok(());
-            }
+            break match self.debug_port_connect(interface, dp) {
```

Comment:
I would suggest trying to phrase this in a way that is easier to read. `break match` is at the very least surprising.

Reply:
The previous version was way more readable :/

Reply:
I'll can pretty easily put the break inside the match if you're uncomfortable with this syntax. In fact I'll probably do that.

Reply:
In general I prefer bounded loops over loop + counter. It's harder to get wrong, less prone to rotting.

This would ideally remain a for loop, and if you want to , you can assign the status to a mutable variable so you can return the last one.

Reply:
I went to do what you're saying, but I'm realizing more and more that I feel very strongly against the for loop. I pose that, in fact, it is the for loop that is (strongly) prone to rotting.

With a for loop we'd have to write something like this:
```rust
let mut error = None;
for retry in 0..num_retries {
    // maybe assign error here, maybe not.
    // maybe return here, maybe not
}

// We're here. Maybe there was an error, maybe not. We'd have to assert that error is Some(err).
```

Essentially we have to deal with the case were `num_retries <= 0`. Now, I know - that _should_ never happen. But code rot happens because of assumptions that later down the line change. I'd rather not make those assumptions.

If you still think this should be a for loop, feel free to push to the branch (You can do that [apparently](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork)), but I personally refuse to write that kind of code.

Reply:
Also, I'd match rather keep the break outside, since it makes absolutely sure we never accidentally keep looping. I don't feel that strongly about this one though, so I could change it if you insist.